samples/XAMLCatalog/XAMLCatalog/UILabelViewController.m

```diff
@@ -0,0 +1,125 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "UILabelViewController.h"
+
+static const CGFloat c_originX = 5;
+static const CGFloat c_originY = 8;
+static const CGFloat c_width = 300;
+static const CGFloat c_height = 50;
+static const CGFloat c_labelFontSize = 17.0f;
+static const int TAG_SUBVIEW_UILABEL = 1;
+
+@implementation UILabelViewController {
+@private
+    NSMutableArray* _labels;
+}
+
+- (UILabel*)_createUILabelWithColor:(UIColor*)color
+                            text:(NSString *)text
+                            textAlignment:(NSTextAlignment)alignment
+                        lineBreakMode:(UILineBreakMode)lineBreakMode
+                        numnberOfLines:(NSInteger )numnberOfLines {
+    CGRect frame = CGRectMake(c_originX, c_originY, c_width, c_height);
+    UILabel* label = [[UILabel alloc] initWithFrame:frame];
+    label.textColor = color;
+    label.text = text;
+    label.textAlignment = alignment;
+    label.lineBreakMode = lineBreakMode;
+    label.numberOfLines = numnberOfLines;
+    return label;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    // creating the text Fields
+    _labels = [[NSMutableArray alloc] init];
+
+    // Row 0. password with number keyboard with Round border
+    [_labels addObject:[self _createUILabelWithColor:[UIColor blackColor]
+                                                 text:@"wordWrap test, You should see this string is wrapped around word"
+                                             textAlignment:UITextAlignmentLeft
+                                         lineBreakMode:UILineBreakModeWordWrap
+                                         numnberOfLines:0]];
+
+    [_labels addObject:[self _createUILabelWithColor:[UIColor redColor]
+                                                 text:@"chararacterWrap is not supported, treated as wordWrap"
+                                             textAlignment:UITextAlignmentLeft
+                                         lineBreakMode:UILineBreakModeCharacterWrap
+                                         numnberOfLines:0]];
+
+    [_labels addObject:[self _createUILabelWithColor:[UIColor blueColor]
+                                                 text:@"Clip and wrapping, the string is wrapped and clipped at the end of second line if the string is long enough"
+                                             textAlignment:UITextAlignmentLeft
+                                         lineBreakMode:UILineBreakModeClip
+                                         numnberOfLines:0]];
```

Comment:
it does seem like they should be separate tests.  does '0' get any special handling vs. a specific line value such as '3'?

---
In reply to: [96539835](https://github.com/Microsoft/WinObjC/pull/1701#discussion_r96539835) [](ancestors = 96539835,96056109)